python-client/tests/communications/test_dto_serialization.py

```diff
@@ -0,0 +1,1149 @@
+        unexpected_index_list=[],
+        output_df=None,
+        number_of_perturbed_rows=TEST_SINGLE_TEST_RESULT_NUMBER_OF_PERTURBED_ROWS,
+        actual_slices_size=[],
+        reference_slices_size=[],
+        output_df_id=TEST_SINGLE_TEST_RESULT_UNEXPECTED_OUTPUT_DF_ID,
+    )
+
+    full_single_test_result_dict = failed_single_test_result.dict()
+    assert "passed" in full_single_test_result_dict.keys()
+    for k in SINGLE_TEST_RESULT_NON_MANDANTORY:
+        assert k in full_single_test_result_dict.keys()
+
+    full_single_test_result_dict_without_none = failed_single_test_result.dict(exclude_none=True)
+    assert "passed" in full_single_test_result_dict_without_none.keys()
+    for k in SINGLE_TEST_RESULT_NON_MANDANTORY:
+        if "output_df" != k:
+            assert k in full_single_test_result_dict_without_none.keys()
+
+
+TEST_IDENTIFIER_SINGLE_TEST_RESULTS_ID = 514
+
+
+def test_identifier_single_test_results():
+    with pytest.raises(ValidationError):
+        websocket.IdentifierSingleTestResult(
+            result=websocket.SingleTestResult(passed=True),
+        )
+    with pytest.raises(ValidationError):
+        websocket.IdentifierSingleTestResult(
+            id=TEST_IDENTIFIER_SINGLE_TEST_RESULTS_ID,
+        )
+    result = websocket.IdentifierSingleTestResult(
+        id=TEST_IDENTIFIER_SINGLE_TEST_RESULTS_ID,
+        result=websocket.SingleTestResult(passed=True),
+    )
+    result_dict = result.dict()
+    assert "id" in result_dict.keys()
+    assert TEST_IDENTIFIER_SINGLE_TEST_RESULTS_ID == result_dict["id"]
+    assert "result" in result_dict.keys()
+    result_dict_without_none = result.dict(exclude_none=True)
+    assert "id" in result_dict_without_none.keys()
+    assert "result" in result_dict_without_none.keys()
+    assert "arguments" not in result_dict_without_none.keys()
+
+    result = websocket.IdentifierSingleTestResult(
+        id=TEST_IDENTIFIER_SINGLE_TEST_RESULTS_ID,
+        result=websocket.SingleTestResult(passed=True),
+        arguments=[],
+    )
+    result_dict = result.dict()
+    assert "id" in result_dict.keys()
+    assert TEST_IDENTIFIER_SINGLE_TEST_RESULTS_ID == result_dict["id"]
+    assert "result" in result_dict.keys()
+    result_dict_without_none = result.dict(exclude_none=True)
+    assert "id" in result_dict_without_none.keys()
+    assert "result" in result_dict_without_none.keys()
+    assert "arguments" in result_dict_without_none.keys()
+
+
+def test_test_suite():
+    with pytest.raises(ValidationError):
+        websocket.TestSuite(is_pass=False, results=[], logs="")
+    with pytest.raises(ValidationError):
+        websocket.TestSuite(is_error=True, results=[], logs="")
+    with pytest.raises(ValidationError):
+        websocket.TestSuite(is_pass=False, results=[])
+
+    test_suite = websocket.TestSuite(is_error=True, is_pass=False, logs="")
+    test_suite_dict = test_suite.dict()
+    assert "is_pass" in test_suite_dict.keys()
```

Comment:
it can be just:

```python
for field in ["is_pass", "is_error", "results", "logs"]:
    assert field in test_suite_dict.keys()
```
in terms of assertions it'll give the same result